File: apps/client/lib/features/auth/application/usecases/login_usecase.dart
import '../../domain/entities/auth_session.dart';
import '../../domain/repositories/auth_repository.dart';

class LoginUseCase {
  const LoginUseCase(this._repo);
  final AuthRepository _repo;
  Future<AuthSession> call({required String email, required String password}) =>
      _repo.login(email: email, password: password);
}

File: apps/backend/src/modules/auth/application/usecases/login.usecase.ts
import { Injectable } from '@nestjs/common';
import { AuthRepository } from '../../domain/repositories/auth.repository';
import { AuthSession } from '../../domain/entities/auth-session.entity';

@Injectable()
export class LoginUseCase {
  constructor(private readonly repo: AuthRepository) {}
  execute(input: { email: string; password: string }): Promise<AuthSession> {
    return this.repo.login(input);
  }
}
